src/hcpp-runtime.c

```diff
@@ -246,11 +265,21 @@ void hcpp_entrypoint() {
         fprintf(stderr, "Error in pthread_attr_init\n");
         exit(3);
     }
-    int starting_worker = 1;
-#ifdef HCPP_COMM_WORKER
-    starting_worker = 2;
+
+    for (int i = 1; i < hcpp_context->nworkers; i++) {
+#ifdef HC_COMM_WORKER
+        /*
+         * If running with a thread dedicated to network communication (e.g. through
+         * UPC, MPI, OpenSHMEM) then skip creating that thread as a compute worker.
+         */
+        assert(communication_worker_id > 0);
```

Comment:
Again, same question, whether you are changing communication worker id? If yes then this needs to be discussed why its important